department-of-veterans-affairs/caseflow
# Base Timeslot Visibility on hearingDay Type (CASEFLOW-1504)

Resolves [CASEFLOW-1504](https://vajira.max.gov/browse/CASEFLOW-1504)

### Description
This makes the Timeslots only visible when scheduling for a virtual hearing day (requestType === 'R').

### Acceptance Criteria
- [x] Time slot buttons are only displayed when a user chooses a Virtual hearing day from the Hearing Date dropdown (request_type === "R")
- [x] Time slot buttons are not displayed when a central (requestType === 'C') or Video (requestType === 'V') hearingDay is selected.

### Reviewer Notes
The ACs of this PR are captured in the two lines of JS. Most of the work was changing the rspec tests:
- Create virtual hearing_days (we weren't doing this in build_hearsched).
- Schedule using all three options
    1. Time dropdown for video/central hearing days.
    2. Timeslot button for virtual hearing days.
    3. Custom time modal for virtual hearing days.

### Testing Plan
1. Go to the scheduling screen for a veteran.
2. Choose any RO that is not the "Virtual Hearings" RO.
3. You should not see the timeslots, even if the hearing type dropdown is set to "Virtual".
4. Change the RO to the "Virtual Hearings" RO.
5. You should now see the timeslots whether you have "Virtual" or "Video" hearing type selected.
